pkg/process/metadata/workloadmeta/grpc.go

```diff
@@ -94,6 +102,14 @@ func (l *GRPCServer) Stop() {
 
 // StreamEntities streams Process Entities collected through the WorkloadMetaExtractor
 func (l *GRPCServer) StreamEntities(_ *pbgo.ProcessStreamEntitiesRequest, out pbgo.ProcessEntityStream_StreamEntitiesServer) error {
+	l.streamMutex.Lock()
+	if l.closeExistingStream != nil {
+		l.closeExistingStream()
+	}
+	streamCtx, cancel := context.WithCancel(context.Background())
+	l.closeExistingStream = cancel
+	l.streamMutex.Unlock()
```

Comment:
Should we refactor this into its own function so that we can use a `defer` statement for the mutex unlock?

Reply:
Great suggestion! I refactored it into its own function named `acquireStreamCtx`